### PR TITLE
Pythontex

### PR DESCRIPTION
**Pull request by ndvanforeest**
2nd try. I used your suggestion, but there is the problem that the minted environment takes python as an option at the start, but does not take this at the end. Thus, the check on a bare "minted".

### CODE REVIEW

File: blacken_docs.py [hunk 1]
```diff
@@ -36,6 +36,13 @@
     r'(?P<after>^(?P=indent)\\end{minted}\s*$)',
     re.DOTALL | re.MULTILINE,
 )
+PYTHONS = r'(\{minted\}|\{minted\}\{python\}|\{(pyblock|pycode|pyconsole|pyverbatim)\})'
```
Comment (asottile) [inline blacken_docs.py:39]: you could have two separate ones, `LATEX_START` and `LATEX_END`

File: blacken_docs.py [hunk 2]
```diff
@@ -36,6 +36,13 @@
     r'(?P<after>^(?P=indent)\\end{minted}\s*$)',
     re.DOTALL | re.MULTILINE,
 )
```
Comment (asottile) [inline blacken_docs.py:38]: also should replace your new version instead of leaving the old `LATEX_RE` around

**Comment by asottile** [comment]
feel free to modify this PR in place instead of making a new one, too

File: blacken_docs.py (hunk 1 above)
Comment (ndvanforeest) [reply blacken_docs.py:39]: That is a good idea. Still I am not really happy. How to guarantee that a pycode start is matched with a pycode end? By giving options, the parser will accept a \begin{pycode} a = 3 \end{pyverbatim}. Of course, tex will tackle this, but how to ensure that blacken-docs will not mess up? Actually, I would build the thing with a match for each separate type of environment (but my python skills are not yours, so I might be very much mistaken here.)

Comment (asottile) [reply blacken_docs.py:39]: oh hmm, yeah -- you could use regex backreferences -- if you leave this open I can modify this PR to get that to work and credit you with the patch 👍

**Comment by ndvanforeest** [comment]
Sorry for messing things up. I'leave this PR open.  BTW, I don't really
mind about the credits :-) I just want to use your tool to tidy up the code
in my tex files and my book :-)

**Comment by asottile** [comment]
> Sorry for messing things up. I'leave this PR open. BTW, I don't really mind about the credits :-) I just want to use your tool to tidy up the code in my tex files and my book :-)

really no problem :) thanks for the work so far on this!

**Comment by ndvanforeest** [comment]
Thanks. I ll study your solution, to improve my regexp knowledge.

**Pull request merged**